services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/paatos/PaatosService.kt

```diff
@@ -23,16 +23,55 @@ class PaatosService(
         val pdfData = alluClient.getDecisionPdf(alluId)
 
         val filename = "${event.applicationIdentifier}-paatos.pdf"
-        val path = uploadPaatos(pdfData, filename, hakemus.id)
+        uploadDecision(
+            pdfData,
+            filename,
+            hakemus.id,
+            alluId,
+            event.applicationIdentifier,
+            PaatosTyyppi.PAATOS,
+            PaatosTila.NYKYINEN,
+        )
+    }
+
+    fun saveKaivuilmoituksenToiminnallinenKunto(
+        hakemus: HakemusEntity,
+        event: ApplicationStatusEvent
+    ) {
+        val alluId = hakemus.alluid!!
+        val pdfData = alluClient.getOperationalConditionPdf(alluId)
+
+        val filename = "${event.applicationIdentifier}-toiminnallinen-kunto.pdf"
+        uploadDecision(
+            pdfData,
+            filename,
+            hakemus.id,
+            alluId,
+            event.applicationIdentifier,
+            PaatosTyyppi.TOIMINNALLINEN_KUNTO,
+            PaatosTila.NYKYINEN,
+        )
+    }
+
+    private fun uploadDecision(
```

Comment:
Why there are both `uploadDecision` and `uploadPaatos´ methods? Should this method also be named as `uploadPaatos`?

Reply:
The naming I had made no sense. I renamed `uploadDecision` to `createPaatos` since it does more than just upload.